contentcuration/contentcuration/utils/publish.py

```diff
@@ -226,15 +226,19 @@ def create_bare_contentnode(ccnode, default_language, channel_id, channel_name):
     if ccnode.language or default_language:
         language, _new = get_or_create_language(ccnode.language or default_language)
 
-    duration = None
-    if ccnode.kind_id in [content_kinds.AUDIO, content_kinds.VIDEO]:
-        # aggregate duration from associated files, choosing maximum if there are multiple, like hi and lo res videos
-        duration = ccnode.files.aggregate(duration=Max("duration")).get("duration")
-
     options = {}
     if ccnode.extra_fields and 'options' in ccnode.extra_fields:
         options = ccnode.extra_fields['options']
 
+    duration = None
+    if ccnode.kind_id in [content_kinds.AUDIO, content_kinds.VIDEO]:
+        completion_criteria = options.get("completion_criteria")
+        if completion_criteria and (completion_criteria["model"] == "time" or completion_criteria["model"] == "approx_time"):
```

Comment:
Please use the `le_utils.constants.completion_criteria` constants.
```python
from le_utils.constants import completion_criteria
completion_criteria.TIME
completion_criteria.APPROX_TIME
```

Reply:
Yup, yup!